core/app/[locale]/(default)/product/[slug]/_components/reviews.tsx

```diff
@@ -48,18 +57,34 @@ const ReviewsQuery = graphql(
   [ProductReviewSchemaFragment, PaginationFragment],
 );
 
-const getReviewsData = cache(async (productId: number) => {
+const getReviewsData = cache(async (productId: number, searchParams: Promise<SearchParams>) => {
+  const pageSize = 5;
+  const { [PaginationSearchParamNames.AFTER]: after, [PaginationSearchParamNames.BEFORE]: before } =
+    await searchParams;
+
+  const variables: VariablesOf<typeof ReviewsQuery> = { entityId: productId };
+
+  if (after != null) {
+    variables.first = pageSize;
+    variables.after = after;
+  }
+
+  if (after == null && before != null) {
+    variables.last = pageSize;
+    variables.before = before;
+  }
```

Comment:
Why not just use this `const paginationArgs = before ? { last: limit, before } : { first: limit, after };` like in other queries?

Reply:
I didn't look at other queries 😅 

I'll make this change